infra/gcp/terraform/k8s-infra-gcp-gcve/vsphere/modules/gcp-gcve-project/main.tf

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# Resource Pools
+
+resource "vsphere_resource_pool" "project_resource_pool" {
+  count = var.nr_projects
+
+  name                    = format("${var.project_name}-%03d", count.index + 1)
+  parent_resource_pool_id = var.vsphere_resource_pool_id
+}
+
+# Folders
+
+resource "vsphere_folder" "project_folder" {
+  count = var.nr_projects
+
+  path          = "${var.vsphere_folder_path}/${format("${var.project_name}-%03d", count.index + 1)}"
+  type          = "vm"
+  datacenter_id = var.vsphere_datacenter_id
+}
+
+# Permissions
+
+# resource "vsphere_entity_permissions" "permission_resource_pool_project" {
```

Comment:
Should we add a TODO here?

Reply:
Removing it for now. Have to follow-up on it once we have a job to work with.

TL/DR is: the terraform plugin has a bug which leads to we may not be able to use it at all.